oneview/data_source_hypervisor_cluster_profile.go

```diff
@@ -0,0 +1,529 @@
+													Computed: true},
+												"name": {
+													Type:     schema.TypeString,
+													Computed: true},
+												"network_uris": {
+													Type:     schema.TypeList,
+													Computed: true,
+													Elem: &schema.Schema{
+														Type: schema.TypeString},
+												},
+												"virtual_switch_ports": {
+													Type:     schema.TypeList,
+													Computed: true,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"action": {
+																Type:     schema.TypeString,
+																Computed: true},
+															"dhcp": {
+																Type:     schema.TypeBool,
+																Computed: true},
+															"ip_address": {
+																Type:     schema.TypeString,
+																Computed: true},
+															"subnet_mask": {
+																Type:     schema.TypeString,
+																Computed: true},
+															"virtual_port_purpose": {
+																Type:     schema.TypeList,
+																Computed: true,
+																Elem: &schema.Schema{
+																	Type: schema.TypeString},
+															},
+														},
+													}},
+												"vlan": {
+													Type:     schema.TypeString,
+													Computed: true},
+											}},
+									},
+									"virtual_switch_type": {
+										Type:     schema.TypeString,
+										Computed: true},
+									"virtual_switch_uplinks": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"action": {
+													Type:     schema.TypeString,
+													Computed: true},
+												"active": {
+													Type:     schema.TypeBool,
+													Computed: true},
+												"mac": {
+													Type:     schema.TypeString,
+													Computed: true},
+												"name": {
+													Type:     schema.TypeString,
+													Computed: true},
+												"vmnic": {
+													Type:     schema.TypeString,
+													Computed: true},
+											},
+										},
+									},
+								},
+							}},
+					}}},
+			"hypervisor_cluster_uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"hypervisor_host_profile_uris": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"hypervisor_manager_uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"hypervisor_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"ip_pools": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString},
+				Set: schema.HashString,
+			},
+
+			"mgmt_ip_settings_override": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"modified": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
If you want to support Get by name and get by URI then make name and URI both as optional. Hope this is what you are trying here.